liesel/model/nodes.py

```diff
@@ -372,6 +372,69 @@ def var(self) -> Var | None:
         """The variable the node is part of."""
         return self._var
 
+    @property
+    def _iloc(self) -> tuple[Node | Var, ...]:
+        input_list: list[Node | Var] = []
+        for input_ in self.inputs:
+            if isinstance(input_, VarValue):
+                if input_.var is None:
+                    raise RuntimeError(f"{input_}.var is None.")
+                input_list.append(input_.var)
+            else:
+                input_list.append(input_)
+
+        return tuple(input_list)
```

Comment:
This is called by `__get_item__`. We are checking that, if `input.var` is `None`, we raise an error, but even if we try to access to an index whose `Var` is not `None`, thus throwing an error when doing that is allowed only because another input's `Var` is `None`. Does it make sense?

Same with the function `_loc`

Reply:
Hi @GianmarcoCallegher, thanks for your comment!
I'm not sure whether I understand the question correctly. Would you go into more detail on what you mean by the sentence „but even if we try to access to an index whose Var is not None, thus throwing an error when doing that is allowed only because another input's Var is None.“? 

Here's what I think:

„try to access to an index whose Var is not None“ In this case, `input_list.append(input_.var)` is called as expected.

Reply:
Maybe a chunk of code will help:

```python
loc = Var(0.0)
scale = Var(1.0)
x = Calc(lambda loc, scale: loc * scale, loc, scale)

x.inputs[0]._var = None

print(x[1])
```

I don't think this should fail since the scale value is not `None`

Reply:
Ah, thank you for the clarification! I understand now. I would actually argue that the current behavior makes sense, for two reasons:

1. In this scenario, `x.inputs[0]` is a `VarValue` instance that does not have an associated variable. In my opinion, this goes against the purpose of the `VarValue` class (wrapping access to a `Var`'s value), and *should* break things.
2. The scenario is created by accessing a private property. This indicates that users should not handle `VarValue._var`, and cannot expect things to work if they do. Things then don't work, which makes sense.

Reply:
I see your point. In particular the second one. Concerning the first one, I think it makes sense but do you think this is the right place to break things then?

Maybe when building the model would be better in my opinion

Reply:
Well, at least I don't think it is bad that things break here in this case. I would argue that we don't even need a separate check for this, because of the second point, i.e. the attribute is private and should not even be handled by the user. This is sort of a general thing; I think it is not feasible to explicitly handle the cases in which users could potentially break the software for themselves if they tried to by manipulating private attributes.

Reply:
That's also a valid point. I'd remove it then

Reply:
When I try to remove it, mypy is unhappy. Would you be ok with leaving the check in there and adding a comment?

Reply:
Sure

Reply:
Done :)